packages/components/src/entities/SamplesEditButton.tsx

```diff
@@ -124,6 +128,10 @@ export const SamplesEditButton: FC<OwnProps & SampleGridButtonProps & RequiresMo
     const showEditParent =
         shouldIncludeMenuItem(SamplesEditButtonSections.EDIT_PARENT, excludedMenuKeys) &&
         hasAnyPermissions(user, [PermissionTypes.Update]);
+    const showMoveToProject =
+        shouldIncludeMenuItem(SamplesEditButtonSections.MOVETOPROJECT, excludedMenuKeys) &&
+        hasAnyPermissions(user, [PermissionTypes.Update]) &&
+        hasProductProjects(moduleContext);
```

Comment:
I went back and forth on this. The check for "does the user have at least one other project with InsertPermission" requires a fetchContainers call with the effectivePermissions included in the response. That would need to happen before the user sees the menu item. I was assuming that is the less frequent case so I decided to show the menu item and then do that check after they click on the item when the modal opens, since we need to do that check and API call there anyway to populate the target container select input. It will show a "you don't have permission to move to any of the available projects" message in the modal for what I assume to be that rare case. Thoughts?

Reply:
Yes, I saw that. I think it's probably fine to leave it shown for now and we'll see how common or uncommon it is that the menu item is present without a user being able to do anything.